p2p/transport/quicreuse/connmgr_test.go

```diff
@@ -92,91 +92,85 @@ func testListenOnSameProto(t *testing.T, enableReuseport bool) {
 // The conn passed to quic-go should be a conn that quic-go can be
 // type-asserted to a UDPConn. That way, it can use all kinds of optimizations.
 func TestConnectionPassedToQUICForListening(t *testing.T) {
-	origQuicListen := quicListen
-	t.Cleanup(func() { quicListen = origQuicListen })
-
-	var conn net.PacketConn
-	quicListen = func(c net.PacketConn, _ *tls.Config, _ *quic.Config) (quic.Listener, error) {
-		conn = c
-		return nil, errors.New("listen error")
+	if runtime.GOOS == "windows" {
+		t.Skip("skipping on windows. Not sure why this fails")
```

Comment:
@marten-seemann Any clue off the top of your head why this fails? https://github.com/libp2p/go-libp2p/actions/runs/5537794470/jobs/10107022242#step:8:4371. 

Presumably this means that on windows after calling `OptimizeConn` the returned conn isn't a `(quic.OOBCapablePacketConn)`

Reply:
Yes, that’s right. It’s using a `basicConn` on Windows: https://github.com/quic-go/quic-go/blob/418b866e3260b65496eb0d4420465e9d11b07e5f/sys_conn.go#L104

I’m becoming more and more convinced that `OptimizeConn` is not the right API. We should change that in a future release.

Reply:
Yeah. I think if Transport has a `WriteTo` that would be better. The problem is I can't treat the conn from OptimizeConn as a normal PacketConn once I give it to the Transport.

Reply:
No, this is fine for now. And the change will be very straightforward